### PR TITLE
fix(docs): Add improvements in examples for PATCH documentation

### PR DESCRIPTION
**Pull request by jjoyce0510**
I've found us having to send code examples directly, so I spent some time cleaning up our docs at least around datasets + Python SDK support. 

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/examples/library/dataset_add_owner_patch.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from datahub.emitter.mce_builder import make_dataset_urn
+from datahub.emitter.rest_emitter import DataHubRestEmitter
+from datahub.metadata.schema_classes import OwnerClass, OwnershipTypeClass
+from datahub.specific.dataset import DatasetPatchBuilder
+
+# Create DataHub Client
+rest_emitter = DataHubRestEmitter(gms_server="http://localhost:8080")
```
Comment (hsheth2) [inline metadata-ingestion/examples/library/dataset_add_owner_patch.py:7]: we should use DataHubGraph everywhere - it's a strict superset of the functionality in the rest emitter

File: metadata-ingestion/examples/library/dataset_field_add_tag_patch.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# Create Dataset URN
+dataset_urn = make_dataset_urn(
+    platform="snowflake", name="fct_users_created", env="PROD"
+)
+
+# Create Dataset Patch to Add + Remove Tag for 'profile_id' column
+patch_builder = DatasetPatchBuilder(dataset_urn)
+patch_builder.for_field("profile_id").add_tag(
+    TagAssociationClass("urn:li:tag:tag-to-add-id")
```
Comment (hsheth2) [inline metadata-ingestion/examples/library/dataset_field_add_tag_patch.py:17]: let's use `make_tag_urn` here

File: metadata-ingestion/examples/library/dataset_add_owner_patch.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+dataset_urn = make_dataset_urn(
+    platform="snowflake", name="fct_users_created", env="PROD"
+)
+
+# Create Dataset Patch to Add + Remove Owners
+patch_builder = DatasetPatchBuilder(dataset_urn)
+patch_builder.add_owner(
+    OwnerClass("urn:li:corpuser:user-to-add-id", OwnershipTypeClass.TECHNICAL_OWNER)
+)
+patch_builder.remove_owner("urn:li:corpGroup:group-to-remove-id")
```
Comment (hsheth2) [inline metadata-ingestion/examples/library/dataset_add_owner_patch.py:19]: we should use of our make_corpuser_urn and corpgroup helpers

File: metadata-ingestion/examples/library/dataset_field_add_tag_patch.py (hunk 2 above)
Comment (jjoyce0510) [reply metadata-ingestion/examples/library/dataset_field_add_tag_patch.py:17]: Not sure sure about this. I think people should not generally be interesting in understanding the IDs and URN structure. I feel like URNs should be the default for logical metadata like Tags, Glossary Terms, and Domains

File: metadata-ingestion/examples/library/dataset_add_owner_patch.py (hunk 3 above)
Comment (jjoyce0510) [reply metadata-ingestion/examples/library/dataset_add_owner_patch.py:19]: Hmm idk - We need to discuss the recommendations on urn creation

File: metadata-ingestion/examples/library/dataset_add_owner_patch.py (hunk 1 above)
Comment (jjoyce0510) [reply metadata-ingestion/examples/library/dataset_add_owner_patch.py:7]: i'm good with this

File: metadata-ingestion/examples/library/dataset_field_add_tag_patch.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/examples/library/dataset_field_add_tag_patch.py:17]: imo this is more straightforward and requires less understanding of urn structure

Users do need to know that this is an urn, but don't need to understand the structure/format of an urn

```suggestion
    TagAssociationClass(make_tag_urn("tag-to-add-id"))
```

**Review by hsheth2 — APPROVED** [review]

**Review by sgomezvillamor — APPROVED** [review]

**Pull request merged**